File: my-app/src/services/MessageService.js
import axios from 'axios';

import axios from 'axios';
import { API_URL } from '../utils/constants';

class MessageService {
  // Get messages between two users
  getMessagesBetweenUsers(user1Id, user2Id) {
    return axios.get(`${API_URL}/messages/${user1Id}/${user2Id}`, {
      headers: {
        'Authorization': `Bearer ${localStorage.getItem('token')}`
      }
    });
  }

  // Mark messages as read
  markMessagesAsRead(fromUserId, toUserId) {
    return axios.put(`${API_URL}/messages/read/${fromUserId}/${toUserId}`, {}, {
      headers: {
        'Authorization': `Bearer ${localStorage.getItem('token')}`
      }
    });
  }

  // Get unread messages for a user
  getUnreadMessages(userId) {
    return axios.get(`${API_URL}/messages/unread/${userId}`, {
      headers: {
        'Authorization': `Bearer ${localStorage.getItem('token')}`
      }
    });
  }
}

export default new MessageService();

File: Backend/src/main/java/com/example/Backend/service/MessageService.java
package com.example.Backend.service;

import com.example.Backend.model.Message;
import com.example.Backend.repository.MessageRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;

@Service
public class MessageService {

    @Autowired
    private MessageRepository messageRepository;

    // Save a new message
    public Message saveMessage(Message message) {
        if (message.getTimestamp() == null) {
            message.setTimestamp(LocalDateTime.now());
        }
        return messageRepository.save(message);
    }

    // Get all messages between two users
    public List<Message> getMessagesBetweenUsers(String user1Id, String user2Id) {
        return messageRepository.findMessagesBetweenUsers(user1Id, user2Id);
    }

    // Mark messages as read
    public void markMessagesAsRead(String fromUserId, String toUserId) {
        List<Message> messages = messageRepository.findMessagesBetweenUsers(fromUserId, toUserId);
        messages.forEach(message -> {
            if (message.getTo().equals(toUserId) && !message.isRead()) {
                message.setRead(true);
                messageRepository.save(message);
            }
        });
    }

    // Get unread messages for a user
    public List<Message> getUnreadMessages(String userId) {
        return messageRepository.findByToAndReadFalse(userId);
    }

    // Get all messages for a user (sent or received)
    public List<Message> getAllMessagesForUser(String userId) {
        return messageRepository.findAllMessagesForUser(userId);
    }
}